server/resp/src/test/java/org/infinispan/server/resp/RespBlockingCommandsTest.java

```diff
@@ -0,0 +1,146 @@
+package org.infinispan.server.resp;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.infinispan.server.resp.test.RespTestingUtil.createClient;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.server.resp.test.RespTestingUtil;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import io.lettuce.core.KeyValue;
+import io.lettuce.core.RedisCommandExecutionException;
+import io.lettuce.core.api.async.RedisAsyncCommands;
+import io.lettuce.core.api.sync.RedisCommands;
+
+/**
+ * Class for blocking commands tests.
+ *
+ * @author Vittorio Rigamonti
+ * @since 15.0
+ */
+@Test(groups = "functional", testName = "server.resp.RespBlockingCommandsTest")
+public class RespBlockingCommandsTest extends SingleNodeRespBaseTest {
```

Comment:
I would also like it if we could test with different value types configured.

Reply:
what do you mean with different value types?

Reply:
Oh, sorry, I could swear I wrote "value media-type" :sweat_smile: Just to check how the listener would behave.

Reply:
I've added a RespBlockingMediaTypes.java. I've commented the TEXT/PLAIN case since it fails. I don't know if we need it though

Reply:
Let me run it here to see if I spot something.

Reply:
Also fails with Protostream.